### ISSUE TITLE
Unify molecule graph data

### ISSUE THREAD

**Issue by nd-02110114** [deepchem/deepchem#1942]
Currently, I seem that there are two data classes, `ConvMol` and `WeaveMol` for a molecule graph and these classes are designed for each specific model (like `ConvMol` <- `GraphConvModel`) 

I think that this situation is not good and we should unify one data class for molecule graph. Generally, all models receive the same graph data class like DGL or PyG. This design leads to make codes more readable and maintainable. 

I recommend DeepChem has only one molecule graph data and all GNN/GCN models receive it. And, current graph data is too complicated, so I also recommend the graph data is simpler like PyG. The PyG design is from [here](https://pytorch-geometric.readthedocs.io/en/latest/notes/introduction.html?highlight=data#data-handling-of-graphs).

**Comment by nd-02110114** [comment]
At the starting point, it may be better to try to use this class like PyG with current DeepChem's featurizer or DiskDataset in the JAXChem project. 
If this works well in the JAXChem project, we can migrate this refactor into DeepChem.

**Comment by rbharath** [comment]
I quite like this suggestion! Ideally our Graph class should be easily interoperable with PyG's class (perhaps a `Graph.to_pytorch_geometric_graph()` method)

This will take a little bit of work to do right though. It would be great to field test this idea in Jaxchem and then upstream into DeepChem.

**Comment by peastman** [comment]
I've also wondered whether we really need different featurizers for them.  Having a unified representation for all graph models would be very nice.

**Comment by nd-02110114** [comment]
> This will take a little bit of work to do right though.

Absolutely... If the big refactor about he DiskDataset class is needed to accept the new graph data, this modification may not be realistic.

**Comment by peastman** [comment]
While we're unifying our molecule representation, we should consider whether we can make it get saved as a numpy structured datatype.  Currently they get stored in Datasets as ConvMol objects which have to be pickled.  That makes loading and saving DiskDatasets much slower than it would be otherwise.

**Comment by rbharath** [comment]
+1 to that idea! That would making loading/saving large datasets much nicer
